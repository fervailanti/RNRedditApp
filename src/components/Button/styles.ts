import { StyleSheet } from 'react-native'

export const styles = (color: string) =>
  StyleSheet.create({
    button: {
      margin: 4,
      paddingHorizontal: 6,
      paddingVertical: 4,
      borderWidth: 2,
      borderRadius: 8,
      flexDirection: 'row',
      alignItems: 'center',
      borderColor: color,
    },
    text: {
      paddingHorizontal: 2,
      fontSize: 14,
      fontWeight: '600',
      textTransform: 'capitalize',
      color: color,
    },
  })
