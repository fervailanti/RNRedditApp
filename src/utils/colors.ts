export const colors = {
  primary: '#FF4400',
  black: '#000000',
  white: '#FFFFFF',
  gray: {
    dark: '#666666',
    light: '#999999',
  },
}
