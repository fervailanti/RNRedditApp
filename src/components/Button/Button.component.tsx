import { Icon } from '@components'
import { colors } from '@utils/colors'
import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import { styles } from './styles'

type ButtonProps = {
  children?: string
  color?: string
  icon?: string
  onPress: () => void
}

const Button: React.FC<ButtonProps> = ({
  color = colors.primary,
  icon,
  onPress,
  children,
}) => {
  return (
    <TouchableOpacity style={styles(color).button} onPress={onPress}>
      {!!icon && <Icon color={color} name={icon} />}
      {!!children && <Text style={styles(color).text}>{children}</Text>}
    </TouchableOpacity>
  )
}

export default Button
