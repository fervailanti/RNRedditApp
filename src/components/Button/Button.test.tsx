import { fireEvent, render } from '@testing-library/react-native'
import { colors } from '@utils/colors'
import React from 'react'
import 'react-native'
import Button from './Button.component'

const props = {
  children: 'Test Button',
  color: colors.primary,
  icon: 'close',
  onPress: () => {},
}

describe('Components > Button', () => {
  it('<Button {...props} /> should render correctly', () => {
    const { toJSON } = render(<Button {...props} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('Given elements inside Button should work as expected', () => {
    const { getByText } = render(<Button {...props} />)
    const button = getByText('Test Button')
    expect(button).toBeTruthy()
    fireEvent.press(button)
  })
})
