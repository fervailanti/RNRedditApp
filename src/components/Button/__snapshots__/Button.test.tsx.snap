// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Button <Button {...props} /> should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#FF4400",
      "borderRadius": 8,
      "borderWidth": 2,
      "flexDirection": "row",
      "margin": 4,
      "opacity": 1,
      "paddingHorizontal": 6,
      "paddingVertical": 4,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#FF4400",
          "fontSize": 16,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ó°…–
  </Text>
  <Text
    style={
      Object {
        "color": "#FF4400",
        "fontSize": 14,
        "fontWeight": "600",
        "paddingHorizontal": 2,
        "textTransform": "capitalize",
      }
    }
  >
    Test Button
  </Text>
</View>
`;
